docs/src/kernel/intel_tdx.md

```diff
@@ -0,0 +1,74 @@
+
+Intel TDX ensures data integrity, confidentiality, and authenticity, which empowers engineers and tech professionals to create and maintain secure systems, enhancing trust in virtualized environments.
+
+For more information, please refer to [Intel TDX website](https://www.intel.com/content/www/us/en/developer/tools/trust-domain-extensions/overview.html).
+
+## Intel TDX environment preparation
+
+Please make sure your server supports Intel TDX.
+
+See [this guide](https://github.com/canonical/tdx/tree/noble-24.04?tab=readme-ov-file#4-setup-host-os) or other materials to enable Intel TDX in host OS.
+
+To verify the TDX host status, you can type:
+
+```bash
+dmesg | grep "tdx"
+```
+
+The following result is an example:
+
+```bash
+[    0.000000] Linux version 6.2.0-tdx.v1.8.mvp10.el8.x86_64 (root@9e8feef4a62e) (gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-15), GNU ld version 2.30-117.el8) #1 SMP PREEMPT_DYNAMIC Tue Apr  4 03:58:57 UTC 2023
+[    0.000000] Command line: BOOT_IMAGE=(hd0,gpt2)/vmlinuz-6.2.0-tdx.v1.8.mvp10.el8.x86_64 root=/dev/mapper/rhel-root ro crashkernel=auto resume=/dev/mapper/rhel-swap rd.lvm.lv=rhel/root rd.lvm.lv=rhel/swap rhgb quiet numa_balancing=disable nouveau.modeset=0 tdx_host=on
+[    2.669909] Kernel command line: BOOT_IMAGE=(hd0,gpt2)/vmlinuz-6.2.0-tdx.v1.8.mvp10.el8.x86_64 root=/dev/mapper/rhel-root ro crashkernel=auto resume=/dev/mapper/rhel-swap rd.lvm.lv=rhel/root rd.lvm.lv=rhel/swap rhgb quiet numa_balancing=disable nouveau.modeset=0 tdx_host=on
+[    2.670044] Unknown kernel command line parameters "rhgb BOOT_IMAGE=(hd0,gpt2)/vmlinuz-6.2.0-tdx.v1.8.mvp10.el8.x86_64 tdx_host=on", will be passed to user space.
+[    5.206655] tdx: BIOS enabled: private KeyID range [32, 64)
+[   12.272499] usb usb1: Manufacturer: Linux 6.2.0-tdx.v1.8.mvp10.el8.x86_64 xhci-hcd
+[   12.274051] usb usb2: Manufacturer: Linux 6.2.0-tdx.v1.8.mvp10.el8.x86_64 xhci-hcd
+[   12.601106]     BOOT_IMAGE=(hd0,gpt2)/vmlinuz-6.2.0-tdx.v1.8.mvp10.el8.x86_64
+[   12.601109]     tdx_host=on
+[   17.377863] tdx: SEAMCALL failed: leaf 254, error 0xc000050500000000.
+[   17.377873] tdx: TDDEBUGCONFIG isn't supported.
+[   17.401371] tdx: TDX module: atributes 0x0, vendor_id 0x8086, major_version 1, minor_version 5, build_date 20230130, build_num 456
+[   17.401378] tdx: TDX module: features0: fbf
+[   20.507284] tdx: 1050629 pages allocated for PAMT.
+[   20.507296] tdx: TDX module initialized.
+[   20.507310] kvm_intel: tdx: max servtds supported per user TD is 1
+[   20.507320] kvm_intel: tdx: live migration supported
+```
+
+`TDX module initialized` means TDX module is loaded successfully.
+
+## Build and run Asterinas
+
+1. Download the latest source code.
+
+```bash
+git clone https://github.com/asterinas/asterinas
+```
+
+2. Run a Docker container as the development environment.
+
+```bash
+docker run -it --privileged --network=host --device=/dev/kvm -v asterinas:/root/asterinas asterinas/asterinas:0.4.2_tdx
+```
```

Comment:
Since a new docker image is used, the file `tools/bump_version.sh` should also be updated. The function `update_image_versions` should also cover the TDX image.

Reply:
Seems that the current implementation is just enough to handle the TDX Docker image name. :smile: